src/home/homework.c

```diff
@@ -1909,6 +1956,13 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 return r;
 
+        dbus_flags_env = getenv("SYSTEMD_HOMEWORK_DBUS_FLAGS");
```

Comment:
hmm, these have nothing really to do with dbus.. if we'd another form of IPC we'd still have them.

frankly, i'd prefer if we'd not pass them as single flags env var, but as indiviudal flags field.

or in other words, translate the flags integer into a bunch of env vars already before even invoking home_start_work() and instead of passing an integer as last arg, just specify a char**, i.e. our usual strv string arrays as last arg, with the env var.

Reply:
Hmm I don't like the idea of passing along an array of env vars into `home_start_work`... We'd need to allocate it somewhere (where? in the immediate context of home_start_work? when parsing dbus?), and deal with ownership (how?), when the underlying data is really just a number.

We could turn it into env vars right before we exec homework, though, I think that makes a bit more sense. I'm still not a huge fan of it though, because we'd be turning a number -> a bunch of env vars, and then parsing a bunch of env vars back into the same number; lot's of code duplication. What's the issue w/ just passing the number into homework?

Reply:
perhaps we should pass the number across via JSON thru an internal field like we do w/ the fdmap, instead of an env var?